src/safeds/data/tabular/containers/_time_series.py

```diff
@@ -0,0 +1,881 @@
+        --------
+        >>> from safeds.data.tabular.containers import Table, TimeSeries
+        >>> table = Table({"date": ["01.01", "01.02", "01.03", "01.04"], "f1": ["a", "b", "c", "a"], "t": [1,2,3,4]})
+        >>> timeseries = TimeSeries._from_table_to_time_series(table, "t", "date", ["f1"])
+        """
+        if feature_names is None:
+            feature_names = table.column_names
+            if time_name in feature_names:
+                feature_names.remove(time_name)
+            if target_name in feature_names:
+                feature_names.remove(target_name)
+        tagged_table = TaggedTable._from_table(table=table, target_name=target_name, feature_names=feature_names)
+        # check if time column got added as feature column
+        return TimeSeries._from_tagged_table(tagged_table=tagged_table, time_name=time_name)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Dunder methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def __init__(
+        self,
+        data: Mapping[str, Sequence[Any]],
+        target_name: str,
+        time_name: str,
+        feature_names: list[str] | None = None,
+    ):
+        """
+        Create a tagged table from a mapping of column names to their values.
+
+        Parameters
+        ----------
+        data : Mapping[str, Sequence[Any]]
+            The data.
+        target_name : str
+            Name of the target column.
+        time_name : str
+            Name of the time column
+        feature_names : list[str] | None
+            Names of the feature columns. If None, all columns except the target column are used.
+
+        Raises
+        ------
+        ColumnLengthMismatchError
+            If columns have different lengths.
+        ValueError
+            If the target column is also a feature column.
+        ValueError
+            If no feature columns are specified.
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import TaggedTable
+        >>> table = TaggedTable({"a": [1, 2, 3], "b": [4, 5, 6]}, "b", ["a"])
+        """
+        _data = Table(data)
+
+        # time sollte nicht in feature names sein auch wenn feature_names none ist
+        if feature_names is None:
+            feature_names = _data.column_names
+            if time_name in feature_names:
+                feature_names.remove(time_name)
+            if target_name in feature_names:
+                feature_names.remove(target_name)
+
+        # Validate inputs
+        super().__init__(data, target_name, feature_names)
+        if time_name in feature_names:
+            raise ValueError(f"Column '{time_name}' can not be time and feature column.")
+        if time_name not in (_data.column_names):
+            raise ValueError(f"Column '{time_name}' must exist in the table.")
```

Comment:
This should be an `UnknownColumnNameError` and it should be mentioned in the docstring

Reply:
added